index.d.ts

```diff
@@ -1,288 +1,591 @@
-    toggleExpanded?: () => any
-  }
+/* #region useGroupBy */
+export function useGroupBy<D extends object = {}>(hooks: Hooks<D>): void
+export namespace useGroupBy {
+  const pluginName = 'useGroupBy'
+}
 
-  export interface UseSortbyOptions {
-    sortByFn?: SortByFn
-    manualSorting?: boolean
-    disableSorting?: boolean
-    defaultSortDesc?: boolean
-    disableMultiSort?: boolean
-  }
+export type UseGroupByOptions<D extends object> = Partial<{
+  manualGroupBy: boolean
+  disableGrouping: boolean
+  aggregations: Record<string, AggregatorFn<D>>
+  groupByFn: (rows: Row<D>[], columnId: IdType<D>) => Record<string, Row<D>>
+}>
+
+export interface UseGroupByHooks<D extends object> {
+  getGroupByToggleProps: ((
+    header: HeaderGroup<D>,
+    instance: TableInstance<D>
+  ) => object)[]
+}
 
-  export interface UseSortbyState<D> {
-    sortBy?: SortingRules<D>
-  }
+export interface UseGroupByState<D extends object> {
+  groupBy: IdType<D>[]
+}
 
-  export interface TableInstance<D = {}> extends TableOptions<D> {
-    hooks: Hooks
-    rows: Row<D>[]
-    columns: EnhancedColumn<D>[]
-    headerGroups: HeaderGroup<D>[]
-    headers: HeaderGroup<D>[]
-    getTableProps: (userProps?: any) => any
-    getRowProps: (userProps?: any) => any
-    prepareRow: (row: Row<D>) => any
-  }
+export type UseGroupByColumnOptions<D extends object> = Partial<{
+  aggregate: Aggregator<D> | Aggregator<D>[]
+  Aggregated: Renderer<CellProps<D>>
+  disableGrouping: boolean
+  groupByBoundary: boolean
+}>
+
+export interface UseGroupByInstanceProps<D extends object> {
+  rows: Row<D>[]
+  preGroupedRows: Row<D>[]
+  toggleGroupBy: (columnId: IdType<D>, toggle: boolean) => void
+}
 
-  export interface TableOptions<D = {}> {
-    data: D[]
-    columns: HeaderColumn<D>[]
-    state: State<D>
-    debug?: boolean
-    loading: boolean
-    defaultColumn?: Partial<Column<D>>
-  }
+export interface UseGroupByColumnProps<D extends object> {
+  canGroupBy: boolean
+  isGrouped: boolean
+  groupedIndex: number
+  toggleGroupBy: () => void
+}
+
+export interface UseGroupByHeaderProps<D extends object> {
+  getGroupByToggleProps: (props?: object) => object
+}
+
+export interface UseGroupByRowProps<D extends object> {
+  isAggregated: boolean
+  groupByID: IdType<D>
+  groupByVal: string
+  values: Record<IdType<D>, AggregatedValue>
+  subRows: Row<D>[]
+  depth: number
+  path: IdType<D>[]
+  index: number
+}
+
+export interface UseGroupByCellProps<D extends object> {
+  isGrouped: boolean
+  isRepeatedValued: boolean
+  isAggregated: boolean
+}
+
+export type DefaultAggregators =
+  | 'sum'
+  | 'average'
+  | 'median'
+  | 'uniqueCount'
+  | 'count'
+
+export type AggregatorFn<D extends object> = (
+  columnValues: CellValue[],
+  rows: Row<D>[]
+) => AggregatedValue
+export type Aggregator<D extends object> =
+  | AggregatorFn<D>
+  | DefaultAggregators
+  | string // TODO: Check that DefaultAggregators works
+export type AggregatedValue = unknown
```

Comment:
I think that this wants to be `any` rather than `unknown`.  As it is, it propagates that unknown through `row.values[id]` accesses.

Reply:
I still haven't ran into any issues with the `unknown` here. Can you point me to a reproducible case where this is causing problems?